src/views/admin/KlassenManagementView.vue

```diff
@@ -0,0 +1,302 @@
+        value: org.id,
+        title: org.name,
+      }));
+      // Update finalKlassen to show in the table
+      finalKlassen.value = organisationStore.klassen;
+    } else {
+      // Reset selectedKlassen and klassenOptions when Schule is unselected
+      selectedKlassen.value = [];
+      klassenOptions.value = [];
+      organisationStore.totalKlassen = 0;
+      // Fetch all Klassen when no Schule is selected
+      await organisationStore.getAllOrganisationen({ includeTyp: OrganisationsTyp.Klasse });
+      finalKlassen.value = organisationStore.allKlassen;
+    }
+  });
+
+  // Watcher to update finalKlassen when Klassen are selected or unselected
+  watch(selectedKlassen, async (newValue: string[] | null, oldValue: string[] | null) => {
+    if (newValue !== oldValue) {
+      if (newValue && newValue.length > 0) {
+        // Filter finalKlassen to only include the selected Klassen
+        finalKlassen.value = organisationStore.klassen.filter((klasse: Organisation) => newValue.includes(klasse.id));
+      } else if (selectedSchule.value !== null) {
+        // If no Klassen are selected but a Schule is selected, show all Klassen for the selected Schule
+        finalKlassen.value = organisationStore.klassen;
+      } else {
+        // If no Klassen and no Schule are selected, show all Klassen
+        await organisationStore.getAllOrganisationen({ includeTyp: OrganisationsTyp.Klasse });
+        finalKlassen.value = organisationStore.allKlassen;
+      }
+    }
+  });
+
+  // Watcher to search Schulen based on input text
+  watch(searchInputSchulen, async (newValue: string, _oldValue: string) => {
+    if (newValue.length >= 1) {
+      // Fetch Schulen matching the search string when it has 3 or more characters
+      await organisationStore.getAllOrganisationen({ searchString: newValue, includeTyp: OrganisationsTyp.Schule });
+    } else {
+      // Fetch all Schulen when the search string is less than 3 characters
+      await organisationStore.getAllOrganisationen({ includeTyp: OrganisationsTyp.Schule });
+    }
+  });
+
+  // Watcher to search Klassen based on input text
+  watch(searchInputKlassen, async (newValue: string, _oldValue: string) => {
+    if (newValue.length >= 1 && selectedSchule.value !== null) {
+      // Fetch Klassen for the selected Schule matching the search string
+      await organisationStore.getKlassenByOrganisationId(selectedSchule.value, newValue);
+      finalKlassen.value = organisationStore.klassen;
+    } else if (selectedSchule.value !== null) {
+      // Fetch all Klassen for the selected Schule when the search string is cleared
+      await organisationStore.getKlassenByOrganisationId(selectedSchule.value);
+      finalKlassen.value = organisationStore.klassen;
+    }
+  });
+
+  const klassenForTable: ComputedRef<Organisation[]> = computed(() => {
+    return finalKlassen.value.filter((organisation: Organisation) => organisation.typ === OrganisationsTyp.Klasse);
```

Comment:
Wieso wird hier noch mal auf Typ Klasse gefiltert? Sollten die Klassen aus dem Store nicht eh immer vom Typ Klasse sein?

Reply:
Dachte Ich mir auch beim Programmieren, wollte ja nur am Ende meine Klassen isolieren und sicherstellen, dass am Ende klar ist, dass nur Klassen in der Tabellle sind :D (Ich mache das weg, unnötig)